addons/test_mail/tests/test_mail_channel.py

```diff
@@ -299,6 +299,39 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
         self.assertEqual(test_chat.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_channel_unfollow_should_also_unsubscribe_the_partner(self):
+        self.assertEqual(self.test_channel.message_channel_ids, self.test_channel)
+        self.assertEqual(self.test_channel.message_partner_ids, self.env['res.partner'])
+        self.assertEqual(self.test_channel.channel_partner_ids, self.env['res.partner'])
+
+        self.test_channel.message_subscribe(self.test_partner.ids)
+        self.assertEqual(self.test_channel.message_channel_ids, self.test_channel)
+        self.assertEqual(self.test_channel.message_partner_ids, self.test_partner)
+        self.assertEqual(self.test_channel.channel_partner_ids, self.env['res.partner'])
```

Comment:
This is a bit overkill testing that `message_subscribe` works, it should already be part of another test, no need to do it here. Just call it to make your setup.

Reply:
Indeed.